framework/src/org/checkerframework/framework/util/QualifierPolymorphism.java

```diff
@@ -392,8 +417,7 @@ protected Void scan(AnnotatedTypeMirror type, Void p) {
 
             assert actualType.getKind() == type.getKind();
             type = (AnnotatedDeclaredType) AnnotatedTypes.asSuper(types, atypeFactory, type, actualType);
-            // TODO: type is null if type is an intersection type
-            // assert type != null;
+
             if (type == null) {
```

Comment:
Not yet.  asSuper returns null erroneously in other cases.  In the new asSuper, it is an ErrorAbort if asSuper returns null and therefore this (and many other) null checks will be removed.  The new asSuper ErrorAborts is how I found this batch of bugs.